pkg/kclient/deployments.go

```diff
@@ -137,13 +138,13 @@ func (c *Client) WaitForPodNotReady(name string) error {
 			if !ok {
 				return errors.New("error getting value from resultchan")
 			}
-			if pod, ok := val.Object.(*corev1.Pod); ok {
-				for _, cond := range pod.Status.Conditions {
-					if cond.Type == "Ready" {
-						if cond.Status == corev1.ConditionFalse {
-							return nil
-						}
-					}
+			if val.Type == apiMachineryWatch.Deleted {
```

Comment:
I we are now watching for deletion of the pod, please also change the method name to reflect the change